app/components/primer/component.rb

```diff
@@ -80,18 +82,35 @@ def validate_arguments(denylist_name: :system_arguments_denylist, **arguments)
         arguments.except!(*denylist.keys.flatten)
       end
 
+      deny_aria_label(tag: tag, arguments: arguments)
+
       arguments
     end
 
     def deny_single_argument(key, help_text, **arguments)
       raise ArgumentError, "`#{key}` is an invalid argument. #{help_text}" \
-        if raise_on_invalid_options? && !ENV["PRIMER_WARNINGS_DISABLED"] && arguments.key?(key)
+        if should_raise_error? && arguments.key?(key)
 
       arguments.except!(key)
     end
 
+    def deny_aria_label(tag:, arguments:)
```

Comment:
We will also want to deny `aria-labelledby`

Reply:
I wanted to start with `aria-label` first to get a feel for it, we can expand it later :)